### PR TITLE
[Merged by Bors] - chore(Algebra/Ring): generalize Injective.isDomain

### PR DESCRIPTION
**Pull request by alreadydone**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by acmepjz** [comment]
😂 Sorry I duplicated your work in #19774. I will close it soon.

**Comment by alreadydone** [comment]
Sorry, I should have linked to my PR #19768 from [my comment](https://github.com/leanprover-community/mathlib4/pull/19614#discussion_r1873935708).

File: Mathlib/Algebra/Ring/Hom/Basic.lean [hunk 1]
```diff
@@ -49,9 +49,7 @@ end Semiring
...
 /-- Pullback `IsDomain` instance along an injective function. -/
-protected theorem Function.Injective.isDomain [Ring α] [IsDomain α] [Ring β] (f : β →+* α)
-    (hf : Injective f) : IsDomain β := by
-  haveI := pullback_nonzero f f.map_zero f.map_one
-  haveI := IsRightCancelMulZero.to_noZeroDivisors α
-  haveI := hf.noZeroDivisors f f.map_zero f.map_mul
-  exact NoZeroDivisors.to_isDomain β
+protected theorem Function.Injective.isDomain [Semiring α] [IsDomain α] [Semiring β] {F}
+    [FunLike F β α] [MonoidWithZeroHomClass F β α] (f : F) (hf : Injective f) : IsDomain β where
+  __ := domain_nontrivial f (map_zero _) (map_one _)
```
Comment (acmepjz) [inline Mathlib/Algebra/Ring/Hom/Basic.lean:54]: Maybe you want to use `RingHom.domain_nontrivial`, i.e. you can write `f.domain_nontrivial` which will be shorter.

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**